core/schema/container.go

```diff
@@ -1109,14 +1169,24 @@ type containerWithMountedDirectoryArgs struct {
 	Path   string
 	Source core.DirectoryID
 	Owner  string `default:""`
+	Expand bool   `default:"false"`
 }
 
 func (s *containerSchema) withMountedDirectory(ctx context.Context, parent *core.Container, args containerWithMountedDirectoryArgs) (*core.Container, error) {
 	dir, err := args.Source.Load(ctx, s.srv)
 	if err != nil {
 		return nil, err
 	}
-	return parent.WithMountedDirectory(ctx, args.Path, dir.Self, args.Owner, false)
+
+	path := args.Path
+	if args.Expand {
+		path, err = expandEnvVar(ctx, parent, args.Path)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
This snippet is very verbose and is repeated a lot. Can we have `expandPath(ctx, parent, expand, path)`?